### ISSUE TITLE
Unexpected header color when theme primary color is set to grey.

### ISSUE THREAD

**Issue by lo5** [h2oai/wave#1257]
#### Wave SDK Version, OS

0.20.0

#### Actual behavior

This makes the header pink:

`ui.theme(name='grey_theme', primary='grey', text='#000000', card='#ffffff', page='#e2e2e2')`

See `--saturatedPrimary`:

![image](https://user-images.githubusercontent.com/6334572/155015071-1abad997-eae7-4531-bdf1-c1a498f96b3d.png)

#### Expected behavior

Header color derivation needs work.

**Comment by mtanco** [comment]
This impacts the side navigation bar too. Buttons with the `primary` attribute are the expected color.

Seems to only be impacting "grey" colors, I upon looking anything with blue, green, reds etc have the `themePrimary` and `saturatedPrimary` colors the same. But grey variations like `55555` and such are differing.

**Comment by mturoci** [comment]
TLDR: This can be special-cased, but we should think about whether saturation for spectrum colors/header/sidebar is a good idea - cc @lo5

### Long story

The reason this is happening for gray only is the fact the grayish colors HSL-wise have 0 hue, 0 saturation and only use lightness (to determine the shade of gray). What Wave does is:

* Look at the saturation of primary color.
* If it's > 50 (primary is a saturated color), decrease it by 30 to achieve desaturated color.
* If it's <= 50 (primary is a desaturated color), increase it by 30 to achieve saturated color.
* Use the primary color saturation for all the spectrum colors.

However, as depicted in the pic below, gray color has always hue 0 (color wheel angle) means 0°, which is red. Saturation (distance from the middle) is also 0, so we increase it by 30 and get light red as a result.

![image](https://user-images.githubusercontent.com/64769322/155138636-9f08171b-3aa2-4815-868d-42425672c77d.png)

**Comment by lo5** [comment]
> we should think about whether saturation for spectrum colors/header/sidebar is a good idea

Explicit color settings for the header/sidebar are fine, too.
As long as it's backwards-compatible.

**Comment by mturoci** [comment]
> Explicit color settings for the header/sidebar are fine, too.

They would be backward compatible, my only concern is that when people will be able to pick any arbitrary color, it might result in disaster considering they are usually not UI developers, but AI people.

What do you think about providing only primary/card/transparent color options without further saturation tweaks?

**Comment by lo5** [comment]
> What do you think about providing only primary/card/transparent color options without further saturation tweaks?

Sure. What would the API look like (i.e. how would the author specify the colors)?

**Comment by mturoci** [comment]
The API would be the same (union type) and the user would get exactly what they asked for (no saturation tweaks).

```ts
  /** Header background color. Defaults to 'primary'. */
  color?: 'card' | 'transparent' | 'primary'
```
